x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -217,6 +231,77 @@ public void testInvalidateApiKeysForApiKeyName() throws InterruptedException, Ex
         assertThat(invalidateResponse.getErrors().size(), equalTo(0));
     }
 
+    public void testInvalidatedApiKeysDeletedByRemover() throws Exception {
```

Comment:
We should add a test that ensures we do _not_ delete api keys with `null` expiration date